### ISSUE TITLE
All day events are created on the correct day, but are moved to the next day after sync.

### ISSUE THREAD

**Issue by tnyeanderson** [SimpleMobileTools/Simple-Calendar#1539]
Thank you for your very hard work on this project. I've used it for years and have noticed an issue that has been occurring for a long time.

Syncing via DAVx5 to Nextcloud.

Create an event on this app, set for all day... for instance on 12/7. When exported, you see it looks correct:
```
BEGIN:VCALENDAR
PRODID:-//Simple Mobile Tools//NONSGML Event Calendar//EN
VERSION:2.0
BEGIN:VEVENT
SUMMARY:Tues 7 test
UID:Caldav-3-2603
X-SMT-CATEGORY-COLOR:-13534004
CATEGORIES:Personal
LAST-MODIFIED:20211111T234155Z
LOCATION:
TRANSP:OPAQUE
DTSTART;VALUE=DATE:20211207
DTEND;VALUE=DATE:20211208
X-SMT-MISSING-YEAR:0
DTSTAMP:20211111T234158Z
STATUS:CONFIRMED
DESCRIPTION:
END:VEVENT
END:VCALENDAR
```

The event is displayed (normally) on the day it was created until DAVx5 performs a sync operation. At this point, in all clients the event is moved to the day after its originally scheduled date (moved from tues to wed in this case). Export shows that DTSTART and DTEND are now one day later each:
```
BEGIN:VCALENDAR
PRODID:-//Simple Mobile Tools//NONSGML Event Calendar//EN
VERSION:2.0
BEGIN:VEVENT
SUMMARY:Tues 7 test
UID:Caldav-3-2603
X-SMT-CATEGORY-COLOR:-13534004
CATEGORIES:Personal
LAST-MODIFIED:19700101T000000Z
LOCATION:
TRANSP:OPAQUE
DTSTART;VALUE=DATE:20211208
DTEND;VALUE=DATE:20211209
X-SMT-MISSING-YEAR:0
DTSTAMP:20211111T234338Z
STATUS:CONFIRMED
DESCRIPTION:
END:VEVENT
END:VCALENDAR

```

As an example, I downloaded Etar, which handles all day events differently. The way Etar handles it works great in all clients before and after sync. Exports from the Etar app of events created in Etar are shown below.

Before sync:
```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:REQUEST
PRODID:-//Etar//ws.xsoh.etar
BEGIN:VEVENT
UID:9ec81a36-381e-4b4e-97d4-b44dedb28547@ws.xsoh.etar
DTSTAMP:20211111T235035Z
SUMMARY:Sun 14 test
DTSTART:20211114T050000Z
DTEND:20211115T050000Z
ORGANIZER;CN=me:mailto:me
END:VEVENT
END:VCALENDAR
```

After sync:
```
BEGIN:VCALENDAR
VERSION:2.0
CALSCALE:GREGORIAN
METHOD:REQUEST
PRODID:-//Etar//ws.xsoh.etar
BEGIN:VEVENT
UID:7e1485b2-8de7-4f4e-a356-e7dfbe091ac0@ws.xsoh.etar
DTSTAMP:20211111T235140Z
SUMMARY:Sun 14 test
DTSTART:20211114T050000Z
DTEND:20211115T050000Z
ORGANIZER;CN=me:mailto:me
END:VEVENT
END:VCALENDAR
```

And a similar event created in Etar, synced, then exported from Simple-Calendar:
```
BEGIN:VCALENDAR
PRODID:-//Simple Mobile Tools//NONSGML Event Calendar//EN
VERSION:2.0
BEGIN:VEVENT
SUMMARY:Tues 16 test
UID:Caldav-3-2602
X-SMT-CATEGORY-COLOR:-13534004
CATEGORIES:Personal
LAST-MODIFIED:19700101T000000Z
LOCATION:
TRANSP:OPAQUE
DTSTART;VALUE=DATE:20211116
DTEND;VALUE=DATE:20211117
X-SMT-MISSING-YEAR:0
DTSTAMP:20211111T234742Z
STATUS:CONFIRMED
DESCRIPTION:
BEGIN:VALARM
DESCRIPTION:Reminder
ACTION:DISPLAY
TRIGGER:-P0DT0H10M0S
END:VALARM
END:VEVENT
END:VCALENDAR
```

> NOTE: The LAST-MODIFIED dates in each of the post-sync Simple-Calendar exports are epoch time. This is kind of weird.

The reason I am creating this issue here is that other clients (Etar) don't seem to have this problem.

Thanks again, I love the Simple Mobile Tools suite!

**Comment by tnyeanderson** [comment]
This may be related to the following in some way:

#1256 
#1521 
#1480

**Comment by tibbi** [comment]
what app and OS version are you using?

**Comment by tnyeanderson** [comment]
Android 11 (CalyxOS 2.11.0)
Simple-Calendar 6.15.4 Pro

Though I had this problem on my old phone as well, which I had last year (would have been Android 10). I only just now had the time to diagnose/debug.

**Comment by azzamsa** [comment]
I have 1.0 Mb worth of events from my previous calendar app. In my case, all of them are imported but lost after I refresh the app to sync.

Only the recurrent event are left, other events are lost.

I did it 5x times, and the behavior is the same and reproducible. Whether I choose to import the old event types or ignore them. 

```
- Simple-Calendar 6.16.2 Pro
- Davx5 v4.0-ose (Oct 14, 2021)
- Android 11 (Samsung One UI 3.1)
```

**Comment by azzamsa** [comment]
@tnyeanderson do you find any workaround? I want to try to import it via Nextcloud Calendar and sync it to Simple-Calendar. Unfortunately, Nextcloud can't import any ics file into existing calendar.

**Comment by tnyeanderson** [comment]
@azzamsa to me, this sounds like a different issue...

Your events seem to disappear, but mine simply get moved to the next day. You may want to open a separate issue with the following extra info:

1. Provide a minimal example .ics file (or entry) which demonstrates the behavior
2. List the exact steps to reproduce the issue
3. Any other relevant information (for instance, what is the origin of this ICS file? Another app? Can you test the imported ics information by exporting it both before and after the sync to see what changes? etc.

It might be the same issue as this one, but to me it doesn't seem like it. It certainly deserves attention as its own issue though :)

**Comment by azzamsa** [comment]
Hi @tnyeanderson.

Yes, I think my imported events also shifted to the next day after importing it to Simple-Calendar. Then it disappears after I hit refresh CalDav.

I tried to import the `old.ics` file via NextCloud then make it sync with Simple-Calendar. To avoid the shifted issue. Unfortunately, NextCloud does not support importing `.ics` into the existing calendar. So, I need to create a new calendar for the old ics file

So my solution:
1. Take a note of the date of some events
2. Import the old `.ics` file into NextClould
3. Sync the new calendar to SC.
4. Close the SC and Sync again. I did this a good couple of times just to make sure no error occurred. Then I compare the synced evens with the date in step one.

It works so far. It has been a day and no sign of shifted day error.

---

Note:

Even the imported event on Simple-Calendar disappears. It is synced to NextCloud. I did the import multiple times because I think the next import will be a susses. So I have many duplicate events in NextCloud. I ended up removing the calendar. 

Luckily, even the NextCloud calendar is removed. All the related events are moved automatically to a local event by Simple-Calendar. Which was nice surprising behavior. So I don't lose any of them.

But, I need to set the event store type to NextCloud for all those events manually. Painful :)

**Comment by eoli3n** [comment]
I face this too.

**Comment by jeremias8** [comment]
Same problem.
Simple-Calendar 6.17.3 Pro
DAVx5 v4.1.1-ose
Android 11 (Android One)

**Comment by evanstucker-hates-2fa** [comment]
I seem to have this issue as well...

Simple Calendar 6.20.0
DAVx5 4.2.3.4-ose
Android 11

I missed my kid's school picture day and two other events before I realized that I wasn't entering these events incorrectly - that this was a bug. My wife is now teasing me about using "weird alternative apps", ha.

I suppose I will either switch to Etar or stop creating "all day" events until this bug is fixed. I'd be happy to help troubleshoot this issue with anyone who wants to work on it.

Despite this issue, I continue to use "weird alternative" FOSS apps forever - open source or GTFO! Keep up the good work!

**Comment by eoli3n** [comment]
I switched to etar, which is much better. Sorry SimpleMobileTools, but you missed it here, Simple Calendar is full of bug, and should not be used at all.

**Comment by Naveen3Singh** [comment]
@tnyeanderson What timezone are you in?

**Comment by Naveen3Singh** [comment]
@tnyeanderson  @eoli3n @jeremias8 @evanstucker-hates-2fa 

If you guys are willing to test, here is a test APK with a possible fix: https://github.com/Naveen3Singh/Simple-Calendar/releases/download/all_day_event_fix/calendar-prepaid-debug.apk ([info](https://github.com/Naveen3Singh/Simple-Calendar/releases/tag/all_day_event_fix))

Lmk if it works (or not)

Thx much.

**Comment by tibbi** [comment]
should be fixed in 6.20.2 released soon

**Comment by tnyeanderson** [comment]
Hi @tibbi which commit fixes this issue?
